analyzers/src/SonarAnalyzer.CSharp/Rules/ObjectShouldBeInitializedCorrectlyBase.cs

```diff
@@ -47,6 +51,8 @@ protected override void Initialize(SonarAnalysisContext context)
             context.RegisterCompilationStartAction(
                 ccc =>
                 {
+                    var isDefaultConstructorSafe = DetermineIfDefaultConstructorIsSafe(context, ccc.Options);
```

Comment:
We should not waste performance investigating `web.config` files when the rule is not active. Let's move this line after the `IsEnabled` check.